webview-ui/src/components/settings/ToolSettings.tsx

```diff
@@ -0,0 +1,170 @@
+import { VSCodeCheckbox } from "@vscode/webview-ui-toolkit/react"
+import { Wrench } from "lucide-react"
+import { HTMLAttributes, useMemo } from "react"
+
+import { useAppTranslation } from "@/i18n/TranslationContext"
+
+import { Section } from "./Section"
+import { SectionHeader } from "./SectionHeader"
+import { SetCachedStateField } from "./types"
+
+type ToolSettingsProps = HTMLAttributes<HTMLDivElement> & {
+	disabledTools?: string[]
+	setCachedStateField: SetCachedStateField<"disabledTools">
+}
+
+// Import the constants from shared/tools.ts
+// Tool display names mapping
+const TOOL_DISPLAY_NAMES: Record<string, string> = {
+	execute_command: "Run commands",
+	read_file: "Read files",
+	fetch_instructions: "Fetch instructions",
+	write_to_file: "Write files",
+	apply_diff: "Apply changes",
+	search_files: "Search files",
+	list_files: "List files",
+	list_code_definition_names: "List definitions",
+	browser_action: "Use a browser",
+	use_mcp_tool: "Use MCP tools",
+	access_mcp_resource: "Access MCP resources",
+	ask_followup_question: "Ask questions",
+	attempt_completion: "Complete tasks",
+	switch_mode: "Switch modes",
+	new_task: "Create new task",
+	insert_content: "Insert content",
+	search_and_replace: "Search and replace",
+	codebase_search: "Codebase search",
+	update_todo_list: "Update todo list",
+}
+
+// Tools that are always available and cannot be disabled
+const ALWAYS_AVAILABLE_TOOLS = [
```

Comment:
the behavior is not specific to ALWAYS_AVAILABLE_TOOLS, they will apply enable/disable to all tools